### PR TITLE
Remove support for passing strategy name to plugins

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Part of #12521 

### Does your PR introduce any breaking changes? If yes, please list them.

Yes. The following is no longer supported:

```py
Trainer(plugins="ddp_spawn")
Trainer(plugins="ddp_find_unused_parameters_false")
Trainer(plugins=DDPStrategy())
```
Instead, use:

```py
Trainer(strategy="ddp_spawn")
Trainer(strategy="ddp_find_unused_parameters_false")
Trainer(strategy=DDPStrategy())
```

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @justusschock @awaelchli @rohitgr7 @borda @kaushikb11 @ninginthecloud

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -229,13 +225,14 @@ def _check_config_and_set_final_flags(
     ) -> None:
         """This method checks:
 
-        1. strategy: strategy and plugin can be set to strategies
+        1. strategy
```
Comment (justusschock) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:228]: can you add some details here on what this method checks for strategies?

**Review by justusschock — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
still draft?

**Comment by awaelchli** [comment]
@rohitgr7 waiting to merge the other one first.

**Review by four4fish — APPROVED** [review]

**Review by Borda — APPROVED** [review]
